api-admin/src/main/java/bio/terra/pearl/api/admin/service/admin/AdminUserExtService.java

```diff
@@ -71,39 +82,126 @@ public AdminUser get(UUID id, String portalShortcode, AdminUser operator) {
     }
-    AdminUser newAdminUser = adminUserService.create(newUser);
-    adminEmailService.sendWelcomeEmail(null, newAdminUser);
-    return newAdminUser;
   }
 
-  public record NewAdminUser(String username, boolean superuser, String portalShortcode) {}
+  @EnforcePortalPermission(permission = "admin_user_edit")
+  public List<String> setPortalUserRoles(
+      PortalAuthContext authContext, UUID adminUserId, List<String> roleNames) {
+
+    authorizeRoles(authContext, roleNames);
+    // looking up the portalAdminUser of the target by the portal from the authContext
+    // confirms they are in a portal the operator has access and permission to edit users
+    PortalAdminUser paUser =
+        portalAdminUserService
+            .findByUserIdAndPortal(adminUserId, authContext.getPortal().getId())
+            .orElseThrow(() -> new NotFoundException("No matching user found"));
+
+    DataAuditInfo auditInfo =
+        DataAuditInfo.builder()
+            .responsibleAdminUserId(authContext.getOperator().getId())
+            .adminUserId(adminUserId)
+            .build();
+    return portalAdminUserRoleService.setRoles(paUser.getId(), roleNames, auditInfo);
+  }
+
+  @SuperuserOnly
+  public void delete(OperatorAuthContext authContext, UUID adminUserId) {
+    DataAuditInfo auditInfo =
+        DataAuditInfo.builder()
+            .responsibleAdminUserId(authContext.getOperator().getId())
+            .adminUserId(adminUserId)
+            .build();
+    adminUserService.delete(adminUserId, auditInfo);
+  }
+
+  /**
+   * removes the user from the specified portal, if the user is in no other portals, also deletes
+   * the user
+   */
+  @EnforcePortalPermission(permission = "admin_user_edit")
+  @Transactional
+  public void deleteInPortal(PortalAuthContext authContext, UUID adminUserId) {
+    DataAuditInfo auditInfo =
+        DataAuditInfo.builder()
+            .responsibleAdminUserId(authContext.getOperator().getId())
+            .adminUserId(adminUserId)
+            .build();
+    // looking up the portalAdminUser of the target by the portal from the authContext
+    // confirms they are in a portal the operator has access and permission to edit users
+    PortalAdminUser paUser =
+        portalAdminUserService
+            .findByUserIdAndPortal(adminUserId, authContext.getPortal().getId())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        "No admin user with id %s in portal %s"
+                            .formatted(
+                                adminUserId.toString(), authContext.getPortal().getShortcode())));
+    portalAdminUserService.delete(paUser.getId(), auditInfo);
+
+    /** if there are no portal users left for the admin, delete the adminUser too */
+    List<PortalAdminUser> portalAdminUsers = portalAdminUserService.findByAdminUser(adminUserId);
+    if (portalAdminUsers.size() == 0) {
+      adminUserService.delete(adminUserId, auditInfo);
```

Comment:
I think we need to handle FKs that still reference this user, i.e. data change records